### ISSUE TITLE
View selection, View All, Reset View

### ISSUE THREAD

**Issue by feiss** [aframevr/aframe-inspector#300]
A very common workflow in 3d packages is to frame the camera to a specific object(s). 

For example, in this blender scene:
![image](https://cloud.githubusercontent.com/assets/359872/17770120/5f0021d2-653c-11e6-8ffe-7aebea073809.png)

If I press the shortcut for "View Selected", the camera jumps to the selected cone:
![image](https://cloud.githubusercontent.com/assets/359872/17770135/687b4070-653c-11e6-8f53-e5da8b5d13e4.png)

The view camera is only **moved**, not rotated or changing its FOV, to a position where the selected object almost fills the entire view. The look-at position of the camera is now the center of the object, so the user can orbit the camera around it.

Another common operation is to "View All", or "Frame All", where the camera moves to frame the whole scene. The look-at position of the camera is now the average center of all objects.

If there's no objects in the scene, the Frame All would reset the camera to a predefined position (look-at at 0,0,0). This could be another available action even if there are objects in the scene: "Reset view"

**Comment by ngokevin** [comment]
Is View Selected appropriate for the Inspector? That sounds useful for a modeling tool, but I don't imagine cases of needing a lock-on feature for individual entities.

**Comment by feiss** [comment]
Yup, modeling is where you take the most of "view selection". But think that in the inspector you have Materials and other components that can change the visual aspect of an entity, and you would want to check it out closely and thoroughly, and "view selection" is useful in this situation.

Also, it's a quick and convenient way of not only moving the camera around the scene, but also of changing its look-at position, useful for orbiting around some areas of the scene.

I use it constantly every minute (after the transform shortcuts, it's the most used key), but that doesn't mean anything, I'd try to ask other people.

**Comment by mingweichan** [comment]
Definitely use this heaps. Been trying to find out how to do this, which is why I landed on this page.

**Comment by Hitschm** [comment]
Same here. I was looking for a way to go to a certain entity in the list where I'm either not exactly sure where it is or where it is hundreds of meters away from the center (where the inspector opens). So if the camera moved to the selected object it would already help a lot to traverse the scene in inspector mode.
Being able to move the inspector camera view trough the scene with keys/mouse would help also. A simple button that would move the camera forward basically. Right now the inspector mode is very restricting when used in larger scenes.

**Comment by etiennepinchon** [comment]
Up! Same problem here! So annoyed by this issue! :( It is hard to focus on one entity at the time. Would be some much smoother if we could zoom toward the current selected entity.

**Comment by HamilcarR** [comment]
Is this little feature still needed ? 
Because I have written some additionnal little functions like this one . 
Maybe I could provide it if I have some time.

**Comment by feiss** [comment]
sure it is!

**Comment by Umbolt** [comment]
Hi, 
I'd like to know how I can open the inspector with a User View or VR view : I mean be able to see the elements as they are displayed in the 360 scene so that I can change the parameters and apreciate directly the changes.  I know there is a command to perform that but I can't find it.
Thanks

**Comment by iangilman** [comment]
For what it's worth, one of these features is kind of available now via the "F" key… It zooms you in to view the current selection. It does change the rotation as well as the position, so that may not be ideal, but it does reposition the "look at" position so you can rotate around the object.
